S2VX.Game/Story/Note/GameNote.cs

```diff
@@ -79,17 +90,21 @@ public class GameNote : S2VXNote {
                     case Key.S:
                     case Key.D:
                     case Key.F:
-                        ScoreInfo.AddScore(Math.Abs(TimingError));
-                        Delete();
+                        ClickNote();
                         break;
                     default:
                         break;
                 }
             }
+
             return false;
         }
 
         protected override void Update() {
+            if (ShouldBeRemoved) {
```

Comment:
Can we leave a comment in the code explaining why this flag is used like this. I think this is a pretty odd usage that should have some documentation/notation explaining what's going on.